pkg/controller/license/trial/trial_controller.go

```diff
@@ -90,23 +90,25 @@ func (r *ReconcileTrials) Reconcile(ctx context.Context, request reconcile.Reque
 	// 2. reconcile the trial license itself
 	trialLicensePopulated := license.IsMissingFields() == nil
 	licenseStatus := r.validateLicense(ctx, license)
+
 	switch {
 	case !trialLicensePopulated && r.trialState.IsTrialStarted():
 		// user wants to start a trial for the second time
-		return r.invalidOperation(ctx, secret, trialOnlyOnceMsg)
+		err = r.invalidOperation(ctx, secret, trialOnlyOnceMsg)
```

Comment:
I think we can keep the early returns here?

```diff
diff --git a/pkg/controller/license/trial/trial_controller.go b/pkg/controller/license/trial/trial_controller.go
index de95a4ea2..5e25e5e58 100644
--- a/pkg/controller/license/trial/trial_controller.go
+++ b/pkg/controller/license/trial/trial_controller.go
@@ -79,7 +79,7 @@ func (r *ReconcileTrials) Reconcile(ctx context.Context, request reconcile.Reque

        validationMsg := validateEULA(secret)
        if validationMsg != "" {
-               return r.invalidOperation(ctx, secret, validationMsg)
+               return reconcile.Result{}, r.invalidOperation(ctx, secret, validationMsg)
        }

        // 1. reconcile trial status secret
@@ -93,16 +93,16 @@ func (r *ReconcileTrials) Reconcile(ctx context.Context, request reconcile.Reque
        switch {
        case !trialLicensePopulated && r.trialState.IsTrialStarted():
                // user wants to start a trial for the second time
-               return r.invalidOperation(ctx, secret, trialOnlyOnceMsg)
+               return reconcile.Result{}, r.invalidOperation(ctx, secret, trialOnlyOnceMsg)
        case !trialLicensePopulated && !r.trialState.IsTrialStarted():
                // user wants to init a trial for the first time
-               return r.initTrialLicense(ctx, secret, license)
+               return reconcile.Result{}, r.initTrialLicense(ctx, secret, license)
        case trialLicensePopulated && !validLicense(licenseStatus):
                // existing license is invalid (expired or tampered with)
-               return r.invalidOperation(ctx, secret, userFriendlyMsgs[licenseStatus])
+               return reconcile.Result{}, r.invalidOperation(ctx, secret, userFriendlyMsgs[licenseStatus])
        case trialLicensePopulated && validLicense(licenseStatus) && !r.trialState.IsTrialStarted():
                // valid license, let's consider the trial started and complete the activation
-               return r.completeTrialActivation(ctx, request.NamespacedName)
+               return reconcile.Result{}, r.completeTrialActivation(ctx, request.NamespacedName)
        case trialLicensePopulated && validLicense(licenseStatus) && r.trialState.IsTrialStarted():
                // all good nothing to do
        }
@@ -175,28 +175,28 @@ func (r *ReconcileTrials) startTrialActivation() error {
        return nil
 }

-func (r *ReconcileTrials) completeTrialActivation(ctx context.Context, license types.NamespacedName) (reconcile.Result, error) {
+func (r *ReconcileTrials) completeTrialActivation(ctx context.Context, license types.NamespacedName) error {
        if r.trialState.CompleteTrialActivation() {
                expectedStatus, err := licensing.ExpectedTrialStatus(r.OperatorNamespace, license, r.trialState)
                if err != nil {
-                       return reconcile.Result{}, err
+                       return err
                }
                _, err = reconciler.ReconcileSecret(ctx, r, expectedStatus, nil)
-               return reconcile.Result{}, err
+               return err
        }
-       return reconcile.Result{}, nil
+       return nil
 }

-func (r *ReconcileTrials) initTrialLicense(ctx context.Context, secret corev1.Secret, license licensing.EnterpriseLicense) (reconcile.Result, error) {
+func (r *ReconcileTrials) initTrialLicense(ctx context.Context, secret corev1.Secret, license licensing.EnterpriseLicense) error {
        if err := r.trialState.InitTrialLicense(ctx, &license); err != nil {
-               return reconcile.Result{}, err
+               return err
        }
-       return reconcile.Result{}, licensing.UpdateEnterpriseLicense(ctx, r, secret, license)
+       return licensing.UpdateEnterpriseLicense(ctx, r, secret, license)
 }

-func (r *ReconcileTrials) invalidOperation(ctx context.Context, secret corev1.Secret, msg string) (reconcile.Result, error) {
+func (r *ReconcileTrials) invalidOperation(ctx context.Context, secret corev1.Secret, msg string) error {
        setValidationMsg(ctx, &secret, msg)
-       return reconcile.Result{}, r.Update(ctx, &secret)
+       return r.Update(ctx, &secret)
 }

 func validLicense(status licensing.LicenseStatus) bool {
```

Reply:
Yeah, it's more readable 👍

68fc6fd